pkg/proxy/nftables/proxier.go

```diff
@@ -1015,6 +1045,104 @@ func isAffinitySetName(set string) bool {
 	return strings.HasPrefix(set, servicePortEndpointAffinityNamePrefix)
 }
 
+// nftElementStorage is an internal representation of nftables map with single string value or set.
+// Currently only supports sets and maps with single string value.
+type nftElementStorage struct {
+	elements      map[string]string
+	leftoverKeys  sets.Set[string]
+	containerType string
+	containerName string
+}
+
+func (c *nftElementStorage) joinKey(k []string) string {
+	return strings.Join(k, " . ")
+}
+
+func (c *nftElementStorage) splitKey(k string) []string {
+	return strings.Split(k, " . ")
+}
+
+// newNFTElementStorage creates an empty nftElementStorage.
+// nftElementStorage.reset() must be called before the first usage.
+func newNFTElementStorage(containerType, containerName string) *nftElementStorage {
+	c := &nftElementStorage{
+		elements:      make(map[string]string),
+		leftoverKeys:  sets.New[string](),
+		containerType: containerType,
+		containerName: containerName,
+	}
+	return c
+}
+
+// reset clears the internal state and flushes the nftables map/set.
+func (c *nftElementStorage) reset(tx *knftables.Transaction) {
```

Comment:
(you should change the variable name you use for `nftElementStorage` objects since it no longer has a `c` anywhere in its name...)